### PR TITLE
feat(models): add openaiStream() OpenAI-compatible SSE formatter

### LINKED ISSUE #514: Add openaiStream() formatter helper for OpenAI-compatible SSE streaming
## Context

The model-access API (#510) ships an OpenAI-compatible `/v1/chat/completions` endpoint so that LangChain.js, the OpenAI SDK, Vercel AI SDK, MCP sampling clients, etc. work against Harper unmodified. For non-streaming responses the shape is straightforward; for streaming, OpenAI uses a very specific SSE delta envelope with a `[DONE]` sentinel.

Rather than have every implementer hand-roll that envelope, ship a single formatter helper in core that wraps a token iterator into the right shape.

## Proposed API

```ts
import { openaiStream } from 'harperdb/streaming';

class ChatCompletions extends Resource {
  async *post(target, body, request) {
    const tokens = scope.models.generateStream(body, {
      model: body.model,
      signal: request.signal,
    });
    yield* openaiStream(tokens, { model: body.model });
  }
}
```

`openaiStream(tokens, { model, id? })` yields `{ event, data, id }` SSE messages whose `data` is a JSON-stringified OpenAI chunk-delta object:

```json
{
  "id": "chatcmpl-...",
  "object": "chat.completion.chunk",
  "created": 1731000000,
  "model": "llama-3.3-70b",
  "choices": [{ "index": 0, "delta": { "content": "..." }, "finish_reason": null }]
}
```

Terminal sentinel: `{ data: '[DONE]' }` (raw — the existing SSE serializer in `contentTypes.ts` handles this since `data: [DONE]\n\n` is exactly what's wanted).

Tool-call deltas (`choices[0].delta.tool_calls`) handled symmetrically when the upstream iterator yields tool-call chunks.

## Acceptance

- [ ] `openaiStream(iter, opts)` is exported from a stable core path.
- [ ] Yielded messages pass through the existing SSE `serializeStream` without modification.
- [ ] Terminal `[DONE]` sentinel emitted.
- [ ] Tool-call chunk deltas formatted correctly.
- [ ] Unmodified OpenAI SDK client (Python and JS) successfully consumes a stream produced by `openaiStream()`.
- [ ] Unmodified LangChain.js / Vercel AI SDK chat-completion stream consumer works.

## Related

- #510 (model-access API and OpenAI-compatible endpoint)
- Soft-blocks the streaming acceptance criterion in FAB-503.

## Out of scope

- Non-chat-completions OpenAI endpoints (Assistants, Files, etc.) — separate work if/when needed.
- Anthropic / Bedrock streaming shapes — those clients accept their native protocols which the respective backends already speak.

### PR DESCRIPTION
**Pull request by heskew**
## Summary

Adds `openaiStream()` (`resources/models/openaiStream.ts`): a formatter that wraps an internal `generateStream()` `AsyncIterable<GenerateChunk>` into OpenAI-compatible `chat.completion.chunk` Server-Sent Events, terminated by the `[DONE]` sentinel. The yielded `{ data }` messages pass through Harper's existing `text/event-stream` serializer (`server/serverHelpers/contentTypes.ts`) unchanged — an object `data` → `data: {json}\n\n`, the sentinel → `data: [DONE]\n\n`.

## Why

Closes #514. This is the SSE formatting helper that #631's `/v1/chat/completions` streaming leg lists as a **hard prerequisite**. Splitting it out keeps #631 to thin protocol-translation Resources over a tested formatter. Part of #510.

## Where to look / design decisions

- **Tool-call mapping (the subtle bit).** Harper backends pre-parse tool-call arguments into objects and yield each complete `ToolCall` (id + name + object-args) exactly once — see `components/openai/index.ts:180-221` (`flushToolCallBuffer` emits only when `id && name`). OpenAI's wire format instead streams `function.arguments` as *string fragments* the client concatenates. So the helper assembles per-id and emits each call's arguments as a single `JSON.stringify`'d blob — emitting incremental fragments would corrupt the client's concatenation (`{"a":1}` + `{"b":2}` → invalid JSON). Consequence: tool calls surface in one delta chunk rather than across many; standard SDKs accept this.
- **SSE shape.** Emits `data:` only — no SSE `event:`/`id:` lines (OpenAI's stream is data-only; the completion id lives inside the JSON payload). Verified against the real serializer in the unit tests.
- **`finish_reason` fallback.** `finishReason ?? (toolAssembly.size ? 'tool_calls' : 'stop')` covers the backend's tail-flush path (`components/openai/index.ts:218-220`), where tool calls are flushed with no explicit `finishReason`.

## Testing

7 unit tests (`unitTests/resources/models/openaiStream.test.js`): content streaming, the `[DONE]` sentinel through the **real** SSE serializer, single + multi tool-call assembly and arg stringification, empty-stream (role + stop), and id stability. Build + lint + format clean.

⚠️ **Not yet smoked end-to-end.** `openaiStream()` has no standalone runtime surface — a real OpenAI-client smoke needs #631's `/v1/chat/completions` endpoint. This stays **draft** until it can be smoked alongside #631; the unit tests assert the exact wire bytes against Harper's serializer in the meantime.

## Cross-model review (HEG step 9)

Both outside-model legs clean, no blockers:
- **Codex** — no actionable correctness issues; ran the build + targeted test itself.
- **Gemini** (`agy`) — all confirmations/suggestions, no blockers; ran mocha; confirmed OpenAI protocol fidelity, edge-case handling, and TypeStrip compliance.

---

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by heskew** [comment]
Rebased onto `main` at v5.1.15 (4212878f0) — no conflicts (both files are net-new). Re-verified locally: `tsc` build emits `dist/resources/models/openaiStream.js` with zero errors on the new files, `oxlint --deny-warnings` clean, and the 7 unit tests pass under both RocksDB and LMDB. Type contract re-checked against the post-#612 `GenerateChunk`/`ToolCall`/`GenerateResult` shapes in `resources/models/types.ts` (every optional field the formatter reads is guarded). Ready for review.

---

🤖 Posted by Claude on Nathan's behalf

File: unitTests/resources/models/openaiStream.sse-http.test.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+// Higher-level coverage for `openaiStream()` (#514, #510): drive its output through
```
[the remark this answers is not in the document]
Comment (heskew) [reply unitTests/resources/models/openaiStream.sse-http.test.js:3]: Addressed in e0fd2ea79 — both openaiStream test files now use plain `node:assert` per the AGENTS.md house style. Repo-wide cleanup + lint enforcement tracked in #1555.

---

🤖 Posted by Claude on Nathan's behalf

**Review by kriszyp — APPROVED** [review]
Reviewed via Claude review-queue — correct SSE framing (data:/[DONE]/delta/finish_reason), lean per-token allocation, no buffering. LGTM.
